### PR TITLE
Changed to types which implement len

### LINKED ISSUE #4613: Recent change to concurrent.futures.wait results in incorrect type errors
A recent [modification](https://github.com/python/typeshed/commit/6e7ac975b499c63056819c5ea9987f089f790049) to concurrent.futures._base.pyi changed the type of the first parameter for `wait` from an `Iterable` to a `Sequence`. @rohitkg98, are you sure this is correct? It generates a bunch of type errors in our code.

Here's a distilled example that demonstrates the problem.

```python
from typing import Any, Dict
from concurrent.futures import Future, wait

# A mapping from the future to its context
futures: Dict[Future[None], Any] = {}

# Generates a type error because futures.keys() is
# a KeysView, which is Iterable but not a Sequence.
wait(futures.keys())
```

### PR DESCRIPTION
**Pull request by rohitkg98**
Using `_Collection` for Python 3 and above.
Using `Container` for Python 2.

Fixes #4613

### CODE REVIEW

**Review by gvanrossum — COMMENTED** [review]
You shouldn’t need the conditionals, because the files are in the /2/ and /3/ sub trees, respectively.

File: stdlib/3/concurrent/futures/_base.pyi [hunk 1]
```diff
@@ -64,9 +64,17 @@ class Executor:
...
+
+if sys.version_info >= (3,):
+    from typing import _Collection
+    def wait(
+        fs: _Collection[Future[_T]], timeout: Optional[float] = ..., return_when: str = ...
+    ) -> Tuple[Set[Future[_T]], Set[Future[_T]]]: ...
+
+else:
+    def wait(
+        fs: Container[Future[_T]], timeout: Optional[float] = ..., return_when: str = ...
```
Comment (JelleZijlstra) [inline stdlib/3/concurrent/futures/_base.pyi:76]: Container only supports `__contains__`, isn't that too wide?

**Review by hauntsaninja — CHANGES_REQUESTED** [review]
Thanks for jumping in to keep improving this!

The conditionals are unfortunately mandated by our current CI, which enforces consistency between certain files. If it were up to me, I think it's probably time to stop making updates to Python 2 stubs unless asked for.

You have, however, run into a pytype error. My guess is that pytype special cases typing.pyi in some way.

I think your best path forward is to just copy over the definition of _Collection from typing.pyi and use it and get rid of the branches. This also lets us get rid of Container (which I think might be a worse type than Iterable for what we want) and should mollify pytype. I'd also add a comment mentioning that the definition of _Collection is just to work around Python 2 / Python 3.5

**Comment by rohitkg98** [comment]
copied the Collection implementation as suggested by @hauntsaninja. Although, couldn't decorate it with `@runtime_checkable`, as that doesn't exist in pytype

**Comment by rchen152** [comment]
pytype has its own copy of the typing stub that includes information on how certain operations change a container's element types. I'd be more than happy to copy over definitions from typing.pyi that typeshed needs (although, I thought that was what [_typeshed](https://github.com/python/typeshed/tree/master/stdlib/2and3/_typeshed) is for?).

Looks like `runtime_checkable` is indeed missing from pytype's typing stub. I'll add it.

**Review by hauntsaninja — APPROVED** [review]
Thanks, this change looks good!

@rchen152 Thanks for offering, but no need to copy _Collection over to pytype's typing.pyi :-) You're right that _typeshed is probably the best place for it if we wanted to make it more reusable within typeshed.

**Comment by JelleZijlstra** [comment]
We don't actually need `@runtime_checkable` here though, since the new class is just a helper for a single stub that can't even be used at runtime.

We can perhaps also remove the `Container` base class since I'm guessing (haven't checked) that `concurrent.futures.wait` doesn't require `__contains__`.

**Comment by hauntsaninja** [comment]
(Jelle's right that we don't need the Container base class; I didn't mention it because I couldn't think of another noun for the class)

**Pull request merged**